ompi/mca/osc/rdma/osc_rdma_lock.h

```diff
@@ -85,10 +87,10 @@ static inline int ompi_osc_rdma_btl_fop (ompi_osc_rdma_module_t *module, struct
             ret = OMPI_SUCCESS;
             ompi_osc_rdma_atomic_complete (module->selected_btl, endpoint, pending_op->op_buffer,
                                            pending_op->op_frag->handle, (void *) pending_op, NULL, OPAL_SUCCESS);
+        } else {
+            /* need to release here because ompi_osc_rdma_atomic_complete was not called */
+            OBJ_RELEASE(pending_op);
```

Comment:
Is this to fix a resource leak issue or "double release" issue? It looks like pending_op will be released anyway before this fix (L91).

Reply:
Double-free. See the comment. In the other path we are calling ompi_osc_rdma_atomic_complete which releases the pending_op. Before this change it could have been released twice if 1 == ret.